### PR TITLE
feat: improve retry functionality

### PR DESCRIPTION
**Pull request by pyrooka**
This PR improves the HTTP retry support to match with the functionality of the other SDK cores.

### CODE REVIEW

File: src/main/java/com/ibm/cloud/sdk/core/http/RetryInterceptor.java [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+
+  // Get the time we should wait before fire the next request in milliseconds.
+  private int getInterval(Response response, Request request) {
+    Integer interval = null;
+
+    String headerVal = response.header("Retry-After");
+
+    if (headerVal != null && !headerVal.equals("")) {
+      try {
+        // TODO: parse as HTTP time, like in Go?
```
Comment (pyrooka) [inline src/main/java/com/ibm/cloud/sdk/core/http/RetryInterceptor.java:83]: https://github.com/IBM/go-sdk-core/blob/main/v5/core/base_service.go#L716

File: src/main/java/com/ibm/cloud/sdk/core/http/RetryInterceptor.java [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+
+    return false;
+  }
+
+  // Calculate the back off time in milliseconds based on the retry count.
+  // This calculation is based on what the go-retryablehttp package does in its
+  // DefaultBackoff function.
+  private int calculateBackoff(int retryCount) {
+    // Exponential interval calculation based on the number of retries.
+    double newInterval = (Math.pow(2, Double.valueOf(retryCount))) * RetryInterceptor.DEFAULT_RETRY_INTERVAL;
```
[the remark this answers is not in the document]
Comment (pyrooka) [reply src/main/java/com/ibm/cloud/sdk/core/http/RetryInterceptor.java:153]: Yes, it starts from 0, so it will be 1-2-4-8.

**Review by padamstx — APPROVED** [review]
LGTM!

**Pull request merged**